models/Metadata/Predefined/Dao.php

```diff
@@ -22,16 +22,25 @@
  *
  * @property \Pimcore\Model\Metadata\Predefined $model
  */
-class Dao extends Model\Dao\PhpArrayTable
+class Dao extends Model\Dao\PimcoreLocationAwareConfigDao
 {
+    use Model\Dao\AutoIncrementTrait;
+
     public function configure()
     {
-        parent::configure();
-        $this->setFile('predefined-asset-metadata');
+        $config = \Pimcore::getContainer()->getParameter("pimcore.config");
+
+        parent::configure([
+            'containerConfig' => $config['metadata']['predefined']['definitions'] ?? [],
```

Comment:
As this is only asset specific I would assume that this should go into the assets namespace similar to
            'containerConfig' => $config['documents']['doc_types']['definitions'] ?? [],

wdyt ?

Reply:
Since metadatas are used in assets only, I would say you are right. I will change the namespace and data structure accordingly.